### PR TITLE
Implementing a hideProfilePicture property on the marker

### PR DESCRIPTION
**Pull request by gahabeen**
Fixing #35 

Tested with non boolean value too and it works.
Didn't manage to add a test with Joi for boolean, I let you check it out if need @Vazerthon ;)

### CODE REVIEW

**Review by Vazerthon — COMMENTED** [review]
Thanks for this, the changes look good and I'm not too worried about adding Joi checks, I think it's safe enough to take a truthy value as true in this case.

I have just realised that I had Prettier settings for single quote and trailing comma in VS Code specific config. I've moved them to `prettier.config.js` now. Would you mind rebasing and running Prettier again please? Sorry, totally my fault!

**Review by Vazerthon — COMMENTED** [review]
Just looking again, I think this needs a little more before it'll work.

`combineDuplicateCoords` is mapping API data into a new shape and will lose the `hideProfilePicture` prop if it exists.
`setUsername` then checks if there are multiple users in each record and re-adds the username for individual users. By default, any combined records won't have a `username` so won't show a picture anyway.
I think the best thing to do would be to rename `setUsername` to `setIndividualUserProps` then add `hideProfilePicture: location.users.length > 1 ? null : location.users[0].hideProfilePicture` at line 38.

File: src/services/user-map-service.js [hunk 1]
```diff
@@ -40,9 +40,9 @@ export const initMap = (hostId) => {
   userMap.addLayer(tiles);
 };
 
-export const addMapMarker = ({ username, lat, lng, caption }) => {
+export const addMapMarker = ({ username, lat, lng, caption, hideProfilePicture }) => {
   const mark = marker([lat, lng], {
-    icon: username ? userIcon(username) : defaultIcon,
+    icon: (username && hideProfilePicture !== false) ? userIcon(username) : defaultIcon,
```
Comment (Vazerthon) [inline src/services/user-map-service.js:45]: I think this can be `icon: (username && !hideProfilePicture) ? userIcon(username) : defaultIcon,`

**Comment by gahabeen** [comment]
> Prettier settings:

Argh I'll check that out ;)

> combineDuplicateCoords is mapping API data into a new shape and will lose the hideProfilePicture prop if it exists.

Sure I'll check that out and update as mentionned!

> I think this can be icon: (username && !hideProfilePicture) ? userIcon(username) : defaultIcon?

I had first set it to `!hideProfilePicture` but somehow I thought it could break if a string (or something else) was input instead. I'm pretty I was wrong, let's put that pack :)

**Comment by Vazerthon** [comment]
I've just set my profile to `{ "usermap_location": { "lat": 55.9485, "lng": -3.2, "caption": "I'm wearing a kilt" }, "hideProfilePicture": "monkey" }` so for testing purposes my pin should no longer show a picture. I'm right in the centre of Edinburgh, Scotland if you can't find me :-)

**Comment by Vazerthon** [comment]
Closing for now

**Pull request closed without merging**